.github/workflows/cd.yml

```
@@ -10,29 +10,32 @@ permissions:
   id-token: write
 
 jobs:
-  # Builds the sdist and wheels on all supported platforms and uploads the resulting
-  # wheels as GitHub artifacts `dev-cibw-*` or `cibw-*`, depending on whether the
-  # workflow is triggered from a PR or a release, respectively.
-  python-packaging:
+  build-sdist:
     name: 🐍 Packaging
-    uses: munich-quantum-toolkit/workflows/.github/workflows/reusable-python-packaging.yml@v1.12
+    uses: munich-quantum-toolkit/workflows/.github/workflows/reusable-python-packaging-sdist.yml@v1.14
+
+  # Builds wheels on all supported platforms using cibuildwheel.
+  # The wheels are uploaded as GitHub artifacts `dev-cibw-*` or `cibw-*`, depending on whether
+  # the workflow is triggered from a PR or a release, respectively.
+  build-wheel:
+    name: 🐍 Packaging
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on:
+          [
+            ubuntu-24.04,
+            ubuntu-24.04-arm,
+            macos-13,
+            macos-14,
+            windows-2022,
+            windows-11-arm,
+          ]
+    uses: munich-quantum-toolkit/workflows/.github/workflows/reusable-python-packaging-wheel-cibuildwheel.yml@v1.14
     with:
-      # Make sure Z3 is available
+      runs-on: ${{ matrix.runs-on }}
       setup-z3: true
-      # 4.13.0 has incorrectly tagged manylinux wheels
       z3-version: 4.12.6
```

```suggestion
      # 4.13.0 has incorrectly tagged manylinux wheels
      z3-version: 4.12.6
```
Can we keep this comment in place?
(and at some point revisit whether newer version of Z3 still have incorrectly tagged wheels)